pkg/reconciler/managed/reconciler.go

```diff
@@ -466,6 +477,10 @@ type ExternalCreation struct {
 	// unless an existing key is overwritten. Crossplane may publish these
 	// credentials to a store (e.g. a Secret).
 	ConnectionDetails ConnectionDetails
+
+	// AdditionalDetails represent any additional details the external client
+	// wants to return about the creation operation that was performed.
+	AdditionalDetails *AdditionalDetails
```

Comment:
Given this is a map type I'd avoid the pointer. The zero value of a map is nil anyway.